src/Grid.php

```diff
@@ -542,9 +542,9 @@ public function addPopup($columnName, $popup = null, $icon = 'caret square down'
      *
      * @return View
      */
-    public function addModalAction($button, $title, \Closure $callback, $args = [])
+    public function addModalAction($button, $title, \Closure $callback, $args = [], $isDisabled = false)
```

Comment:
I belive this is not a per row, not sure what usecase it should solve.

Reply:
Sure, the isDisabled is either boolean or closure, if you pass function($row) { return ($row->get('age') > 12); } you can dynamically set disabled per row as needed. @mvorisek

Reply:
this need a phpdoc `@param` above the method like the `enabled` property phpdoc

https://dev.agiletoolkit.org/demos/tutorial/actions.php?atk_layout_maestro_wizard=3 - UA supports this natively, why do you want to extent this method? Advanced options should not be configureable by a custom parameter. One of my usecases was a different icon for example, should this be addressed by another param? No.

Reply:
addAction button has this feature today and all I do is to allow also addModalAction to use this helpful feature. Not everyone is using UA - a lot are using addActionButton etc. to configure Crud which is helpful. I do not see why this is a problem. Or do you want to delete also the isDisabled feature from addActionButton? That would be a pity

Reply:
creating UA with a seed should be the desired way to solve \*any\* related special settings, I am quite against the proposed changes

Reply:
Sure we could re-work this to flexibly have multiple advanced parameters - I'd say that would be an idea for future improvements. Alternative could be ActionButtons::onHook(HOOK_GET_HTML_TAGS, function ...) similar to 
`$table->onHook(Table\Column::HOOK_GET_HTML_TAGS, function (Table $table, Model $row) { ...`where we return array of icon, disabled, title

Reply:
I do not see why you are against it? There are multiple ways to code, and UA is not always the best way, especially if you want to dynamically adjust Cruds' action buttons in certain views. We should not enforce one single way of coding, but keep room for alternatives, as we all do not have all use cases in mind. 
This PR is nothing else than creating consistency between addActionButton and addModalAction with existing features. I am absolutely against removing isDisabled feature unless UA becomes much more easy to code with less lines of code. And even then there might be a lot of reasons why people do not want to use UA but call addActionButton/ addModalAction directly.

Reply:
@ibelar Any view on this PR and our pro / con discussion here?

Reply:
many params is always a bad design :)

if UA params have to be supported by these methods, it should be accepted as a seed

also missing a test, at least a render demo like https://ui.agiletoolkit.org/demos/tutorial/actions.php?atk_layout_maestro_wizard=3